### PR TITLE
Release 0.3.0

### PR DESCRIPTION
**Pull request by mike0sv**
This is a list of activities that should be done before release announce

### Breaking changes - merge in this PR
- [x] New deployment state management
- [x] New CLI interface: breaking changes in `serve`, `apply-remote`, `build` and `declare`

### Other changes - merge in this PR
- [x] SageMaker deployments
- [x] K8s deployments
- [x] `--quiet` https://github.com/iterative/mlem/pull/448 @mike0sv 

### Docs
- [x] Finish semi-generated API reference PR https://github.com/iterative/mlem.ai/pull/172
- [x] Update Quick Start in this repo
- [x] https://github.com/iterative/mlem.ai/pull/197
- [x] https://github.com/iterative/mlem.ai/pull/188

### Blog post
- [x] blog post re release @aguschin 

### Example-repo - PR in `example-repos-dev`
- [x] update `example-mlem-get-started`

### Update the website code animations
- [x] https://mlem.ai

### MLEM Release
- [x] merge this branch
- [x] release it on PyPi
- [x] merge docs PR
- [x] update http://github.com/iterative/example-mlem-get-started

## Third, things after release

- [x] merge PR in `example-repos-dev` and update `iterative/example-mlem-get-started` https://github.com/iterative/example-repos-dev/pull/139
- [x] remove `xfail` mark from `tests/cli/test_main.py::test_commands_docs_links` https://github.com/iterative/mlem/pull/462

### CODE REVIEW

File: mlem/cli/deployment.py [hunk 1]
```diff
@@ -4,90 +4,154 @@
...
 
 deployment = Typer(
     name="deployment",
-    help="Manage deployments",
+    help="A set of commands to set up and manage deployments",
     cls=mlem_group("runtime", aliases=["deploy"]),
 )
 app.add_typer(deployment)
 
+deploy_run = Typer(
```
Comment (aguschin) [inline mlem/cli/deployment.py:53]: Using thread (Using thread to keep comments under control, this is unrelated to a specific line highlighed)
Can't build docker image with this branch:
```cli
$ mlem declare env docker --registry local --daemon.host localhost docker-builder.mlem
$ mlem build docker --file_conf docker-builder --model rf.mlem --image.name=rf-docker
not enough values to unpack (expected 2, got 1)
```
What's wrong? @mike0sv

File: .github/workflows/check-test-release.yml [hunk 2]
```diff
@@ -7,6 +7,7 @@ on:
 
 env:
   MLEM_TESTS: "true"
+  MLEM_DEBUG: "true"
```
Comment (aguschin) [inline .github/workflows/check-test-release.yml:10]: Another bug:

```shell
$ mlem --tb deployment run heroku app.mlem --model models/rf --app_name mlem-get-started
...
mlem.core.errors.MlemObjectNotFound: MLEM object was not found at `heroku`
```
This should work, judging by your examples [here](https://github.com/iterative/mlem/pull/417#issue-1383952509).

File: mlem/cli/deployment.py (hunk 1 above)
Comment (aguschin) [reply mlem/cli/deployment.py:53]: `--file_conf` is in `$ mlem build docker -h` for me. Don't see `--load` there.

File: .pylintrc [hunk 3]
```diff
@@ -170,7 +170,8 @@ disable=print-statement,
         redefined-builtin, # TODO: https://github.com/iterative/mlem/issues/60
         no-self-use, # TODO: https://github.com/iterative/mlem/issues/60 maybe leave it
         import-outside-toplevel,
-        wrong-import-order # handeled by isort
+        wrong-import-order, # handeled by isort
+        cannot-enumerate-pytest-fixtures # TODO: https://github.com/iterative/mlem/issues/60
```
Comment (aguschin) [inline .pylintrc:174]: Some bug was introduced in this branch. If you try to build docker image and then run it:
```shell
$ mlem build docker docker-builder.mlem --model models/rf --image.name mlem-model --env.daemon.host "" --server fastapi
⏳️ Loading model from models/rf.mlem
🛠 Building MLEM wheel file...
  DEPRECATION: A future pip version will change local packages to be built in-place without first copying to a temporary directory. We recommend you use --use-feature=in-tree-build to test your packages with this new behavior before it becomes the default.
   pip 21.3 will remove support for this functionality. You can find discussion regarding this at https://github.com/pypa/pip/issues/7555.
💼 Adding model files...
🛠 Generating dockerfile...
💼 Adding sources...
💼 Generating requirements file...
🛠 Building docker image mlem-model:latest...
✅  Built docker image mlem-model:latest
(mlem) get-started (release/0.3.0) $ docker run mlem-model:latest                                                                                26.64s
2022-10-13 16:36:45,627 [DEBUG] mlem.core.hooks: Not registerting IsInstanceHookMixin to any Analyzer because it's an abstract class
2022-10-13 16:36:45,632 [DEBUG] mlem.core.hooks: Not registerting RequirementsHook to any Analyzer because it's an abstract class
2022-10-13 16:36:45,632 [DEBUG] mlem.core.hooks: Not registerting AddRequirementHook to any Analyzer because it's an abstract class
2022-10-13 16:36:45,903 [DEBUG] mlem.core.hooks: Not registerting DataHook to any Analyzer because it's an abstract class
2022-10-13 16:36:45,905 [DEBUG] mlem.core.hooks: Registering PrimitiveType to DataAnalyzer
2022-10-13 16:36:45,911 [DEBUG] mlem.core.hooks: Registering OrderedCollectionHook to DataAnalyzer
2022-10-13 16:36:45,911 [DEBUG] mlem.core.hooks: Registering DictTypeHook to DataAnalyzer
2022-10-13 16:36:45,921 [DEBUG] mlem.core.hooks: Not registerting ModelHook to any Analyzer because it's an abstract class
2022-10-13 16:36:46,064 [DEBUG] mlem.core.hooks: Not registerting ImportHook to any Analyzer because it's an abstract class
2022-10-13 16:36:46,065 [DEBUG] mlem.core.hooks: Not registerting ExtImportHook to any Analyzer because it's an abstract class
2022-10-13 16:36:46,066 [DEBUG] mlem.core.hooks: Not registerting LoadAndAnalyzeImportHook to any Analyzer because it's an abstract class
2022-10-13 16:36:46,066 [DEBUG] mlem.core.hooks: Registering PickleImportHook to ImportAnalyzer
Traceback (most recent call last):
  File "/usr/local/bin/mlem", line 5, in <module>
    from mlem.cli import app
  File "/usr/local/lib/python3.9/site-packages/mlem/__init__.py", line 9, in <module>
    from . import api  # noqa
  File "/usr/local/lib/python3.9/site-packages/mlem/api/__init__.py", line 5, in <module>
    from .commands import (
  File "/usr/local/lib/python3.9/site-packages/mlem/api/commands.py", line 38, in <module>
    from mlem.runtime.client import Client
  File "/usr/local/lib/python3.9/site-packages/mlem/runtime/__init__.py", line 1, in <module>
    from .interface import Interface, Signature
ImportError: cannot import name 'Interface' from 'mlem.runtime.interface' (/usr/local/lib/python3.9/site-packages/mlem/runtime/interface/__init__.py)
```

**Comment by aguschin** [comment]
@madhur-tandon, btw this branch have few issues I've reported - they prevent me from finishing GS for build and deploy in [new docs PR](https://github.com/iterative/mlem.ai/pull/188). If you could fix them, that would be awesome!

File: .pylintrc (hunk 3 above)
Comment (mike0sv) [reply .pylintrc:174]: Can't reproduce this :(

Comment (aguschin) [reply .pylintrc:174]: Why `:(`? Maybe this is my local issue and it's one less bug to fix 😄

Comment (aguschin) [reply .pylintrc:174]: Confirmed, it works. Something wrong with my installation then!

File: mlem/cli/deployment.py (hunk 1 above)
Comment (aguschin) [reply mlem/cli/deployment.py:51]: Looks like this wasn't fixed by your last PR @mike0sv. This is not that urgent though. @madhur-tandon, could you please TAL and fix what I've reported in the last message above?

File: .github/workflows/check-test-release.yml (hunk 2 above)
Comment (madhur-tandon) [reply .github/workflows/check-test-release.yml:10]: I don't face this issue, I am using
`mlem --tb deployment run heroku app.mlem --model sk-model --app_name mlem-get-started`
and I face the issue of `FileNotFoundError: [Errno 2] No such file or directory: 'heroku'`
which is expected and is because of me not passing the API key which it tries to determine using
`subprocess.check_output(["heroku", "auth:token"])`
and I don't have the `heroku` CLI installed so...

Can you try this again @aguschin? maybe make sure you are using the `release 0.3` branch, do a
`pip install -e .` again after switching to it.

Comment (aguschin) [reply .github/workflows/check-test-release.yml:10]: FIxed! Thanks

File: mlem/cli/deployment.py (hunk 1 above)
Comment (aguschin) [reply mlem/cli/deployment.py:51]: Thank you @madhur-tandon, indeed, those were issues with my installation. Could you please fix the last 2 items? They still appear for me.

Comment (aguschin) [reply mlem/cli/deployment.py:51]: I've created an issue so we won't forgot addressing it later https://github.com/iterative/mlem/issues/441

**Review by aguschin — APPROVED** [review]
NICE! Congrats everyone! @mike0sv @madhur-tandon 🚀 this is an awesome step in the direction of the awesomeness!
